jobs/sql/extractions/hiv_monthly_reporting.sql

```diff
@@ -96,6 +105,56 @@ set latest_hiv_viral_load_coded	= obs_from_group_id_value_coded_list(latest_hiv_
 update temp_eom_appts t
 set latest_hiv_viral_load	= obs_from_group_id_value_numeric(latest_hiv_viral_load_obs_group,'CIEL','856');
 
+-- latest arv treatment line 
+-- a temp table is loaded with the most recent HIV order information for each row
+-- this is used to update the main temp table
+drop temporary table if exists temp_last_arv_order_id;
+create temporary table temp_last_arv_order_id 
+	select t.monthly_reporting_id , o.order_id, o.date_activated, o.encounter_id  
+	from temp_eom_appts t 
+	inner join 
+		(select o2.order_id, o2.patient_id ,  o2.date_activated, o2.encounter_id  
+		from orders o2 
+		where o2.voided = 0
+		and o2.order_reason = concept_from_mapping('CIEL','138405')
+		order by if(o2.date_stopped is null,0,1) asc, o2.date_activated desc) o on o.patient_id = t.patient_id and o.date_activated < t.reporting_date  
+	group by t.monthly_reporting_id 
+;
+
+update temp_eom_appts t 
+inner join temp_last_arv_order_id tlao on tlao.monthly_reporting_id = t.monthly_reporting_id 
+set latest_arv_regimen_order_id = tlao.order_id,
+	latest_arv_regimen_date = tlao.date_activated,
+	latest_arc_regimen_encounter_id = tlao.encounter_id ;
+
+update temp_eom_appts t 
+set latest_arv_regimen_line = obs_value_coded_list(t.latest_arc_regimen_encounter_id, 'PIH','13115',@locale );
+
+-- latest arv dispensing
+-- a temp table is loaded with the most recent HIV dispensing encounter information for each row
+-- this is used to update the main temp table
+drop temporary table if exists temp_last_arv_dispensing_id;
+create temporary table temp_last_arv_dispensing_id 
+	select t.monthly_reporting_id, e.encounter_id, e.encounter_datetime  
+	from temp_eom_appts t 
+	inner join 
+		(select e2.patient_id, e2.encounter_id, e2.encounter_datetime 
+		from encounter e2
+		inner join obs o on o.voided = 0 and o.encounter_id = e2.encounter_id and  o.concept_id =  concept_from_mapping('PIH','1535')
+			and o.value_coded in (concept_from_mapping('PIH','3013'),concept_from_mapping('PIH','2848'),concept_from_mapping('PIH','13960'))
+		where e2.voided = 0
+		and e2.encounter_type = @hiv_Dispensing_id 
+		order by e2.encounter_datetime desc, e2.encounter_id desc) e on e.patient_id  = t.patient_id and e.encounter_datetime <= t.reporting_date 
```

Comment:
Are you sure this will always return the last date, if there are multiple dispensing encounters in a month?